e2e/cypress/integration/channel/message_shortlinking_spec.js

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+describe('Message', () => {
+    beforeEach(() => {
+        // # Login as "user-1" and go to /
+        cy.apiLogin('user-1');
+
+        cy.visit('/');
+    });
+
+    it('M17451 Channel shortlinking still works when placed in brackets', () => {
+        // # Post shortlink of channel
+        const shortLink = '(~saepe-5)';
+        const longLink = '~doloremque';
+
+        cy.postMessage(shortLink);
+
+        // # Grab element with the long link url and go to the link
+        cy.get('a.mention-link').contains(longLink).click();
```

Comment:
You should probably target the last created post, depending on prior posts, this could find a previous message that includes this link (such as if you run this test multiple times).

Reply:
Not sure if I am understanding correctly, do you suggest something like this as opposed to by reference to 'a.mention-link'? 
```js
        // #grabs all posts in channel, looks for posts containing link, then click on the last link 
        cy.get('.post-message__text').contains(longLink).last().click();
```

Reply:
Actually, I think ` .contains` return the first match instead of an array of them 
```js
cy.get('.post-message__text').last().contains(longLink).click();
```

Reply:
got it, Thanks!